packages/block-editor/src/components/block-list/layout.js

```diff
@@ -24,12 +25,23 @@ export function useLayout() {
 	return useContext( Layout );
 }
 
-export function LayoutStyle( { layout = {}, ...props } ) {
+export function LayoutStyle( { layout = {}, css, ...props } ) {
 	const layoutType = getLayoutType( layout.type );
+	const blockGapSupport = useSetting( 'spacing.blockGap' );
+	const hasBlockGapSupport = blockGapSupport !== null;
 
 	if ( layoutType ) {
-		return <layoutType.save layout={ layout } { ...props } />;
+		if ( css ) {
+			return <style>{ css }</style>;
+		}
```

Comment:
I know this is old, but I'm not sure I understand why we have a `css` prop in this component, why not just use `<style>{css}</style>` instead of `<LayoutStyle css={css} />`? am I missing anything? It seems it has nothing to do with layout.

Reply:
It could totally be refactored. If I remember correctly, the thinking at the time was to have a single entry point for outputting layout styles via the `LayoutStyle` component, as it includes a check that the layout type is valid and made for a little more consistency in the consuming code... the thinking at the time was also that we might further refactor the innards of `LayoutStyle` to not necessarily output a `style` tag, so in theory, having that abstracted into the `LayoutStyle` component could be beneficial.

There's been a fair few changes since this PR was merged, though, and with the recent work on https://github.com/WordPress/gutenberg/pull/52888 and I see https://github.com/WordPress/gutenberg/pull/54466 has just landed to refactor some of the layout style output, I don't think we need to hold onto any of the decisions in this older PR 🙂